### DISCUSSION TITLE
Sign Tron Transaction Issue

### DISCUSSION THREAD

**Discussion by chamid** [trustwallet/wallet-core#2467] [Q&A]
I want to sign a transaction and broadcast it in network
but I got error response 

Here is my Code:
`val tron = WalletHelper.wallet
        val trc20Contract = Tron.TransferTRC20Contract.newBuilder()
            .setOwnerAddress(tron.getAddressForCoin(CoinType.TRON))
            .setContractAddress("Contract")
            .setToAddress("To")
            // 1000
            .setAmount(5.0.toBigDecimal().multiply(BigDecimal.TEN.pow(6)).toBigInteger().toByteString())

        val blockHeaderInfo = param.blockHeaderInfo

        val blockHeader = Tron.BlockHeader.newBuilder()
            .setTimestamp(blockHeaderInfo?.timestamp?: 0L)
            .setTxTrieRoot(ByteString.copyFrom(blockHeaderInfo?.txTrieRoot?.toByteArray()))
            .setParentHash(ByteString.copyFrom(blockHeaderInfo?.parentHash?.toByteArray()))
            .setNumber(blockHeaderInfo?.number?: 0)
            .setWitnessAddress(ByteString.copyFrom(blockHeaderInfo?.parentHash?.toByteArray()))
            .setVersion(blockHeaderInfo?.version?: 0)
            .build()

        val transaction = Tron.Transaction.newBuilder()
            .setTransferTrc20Contract(trc20Contract)
            .setTimestamp(System.currentTimeMillis() + (2 * 60 * 60 * 1000))
            .setBlockHeader(blockHeader)
            .setFeeLimit(1000000000)
            .build()

        val signingInput = Tron.SigningInput.newBuilder()
            .setTransaction(transaction)
            .setPrivateKey(ByteString.copyFrom(tron.getKeyForCoin(CoinType.TRON).data()))

        val output = AnySigner.sign(signingInput.build(), CoinType.TRON, Tron.SigningOutput.parser())

        return output.signature.toByteArray().toHexString(false)`

this is output json of my transaction:

`{
  "raw_data": {
    "contract": [
      {
        "parameter": {
          "type_url": "type.googleapis.com/protocol.TriggerSmartContract",
          "value": {
            "contract_address": "41a614f803b6fd780986a42c78ec9c7f77e6ded13c",
            "data": "a9059cbb000000000000000000000041e2d6c8039fbc2da047642e7017c046aeed65364300000000000000000000000000000000000000000000000000000000004c4b40",
            "owner_address": "41a8e79cf78bd92abd03980adb849483963772cad3"
          }
        },
        "type": "TriggerSmartContract"
      }
    ],
    "expiration": 1659240583776,
    "fee_limit": 1000000000,
    "ref_block_bytes": "3467",
    "ref_block_hash": "0c3058558647abbc",
    "timestamp": 1659204583776
  },
  "signature": [
    "7fc2a815736d0703683fb3bfd5e123a7e479922b671333d653724e68974391ef7b8a50b34a757b2fcbfbc287741d61a0b7fc3663d00f50c96492f902f452054c00"
  ],
  "txID": "b30fe126270db271be11971877ff777b684cc21763a49312e3420945b9d52c40"
}`

I send this to: https://api.trongrid.io/wallet/broadcasttransaction

but I got this response:

`{
    "code": "TAPOS_ERROR",
    "txid": "b30fe126270db271be11971877ff777b684cc21763a49312e3420945b9d52c40",
    "message": "5461706f7320636865636b206572726f722e"
}`

**Comment by vcoolish** [comment]
```
val transaction = Tron.Transaction.newBuilder()
        .setTransferTrc20Contract(trc20Contract)
        .setTimestamp(blockInfo.timestamp)
        .setExpiration(blockInfo.timestamp + (2 * 60 * 60 * 1000))
        .setBlockHeader(blockHeader)
        .setFeeLimit(1000000000)
        .build()
```
